compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/internal/parser/BallerinaParser.java

```diff
@@ -12492,6 +12496,100 @@ private STNode parseVarKeyword() {
         }
     }
 
+    /**
+     * Parse list match pattern.
+     * <p>
+     *     <code>
+     *         list-match-pattern := [ list-member-match-patterns ]
+     *         list-member-match-patterns :=
+     *          match-pattern (, match-pattern)* [, rest-match-pattern]
+     *          | [ rest-match-pattern ]
+     *     </code>
+     * </p>
+     *
+     * @return Parsed list match pattern node
+     */
+    private STNode parseListMatchPattern() {
+        startContext(ParserRuleContext.LIST_MATCH_PATTERN);
+        STNode openBracketToken = parseOpenBracket();
+        List<STNode> matchPatternList = new ArrayList<>();
+        STNode restMatchPattern = null;
+        STNode matchPatternMemberRhs;
+
+        do {
+            STToken nextToken = peek();
+            if (nextToken.kind == SyntaxKind.ELLIPSIS_TOKEN) {
+                restMatchPattern = parseRestMatchPattern();
+                break;
+            }
+            STNode matchPatternListMember = parseMatchPattern();
+            matchPatternList.add(matchPatternListMember);
+            matchPatternMemberRhs = parseListMatchPatternMemberRhs();
+
+            if (matchPatternMemberRhs != null) {
+                matchPatternList.add(matchPatternMemberRhs);
+            }
+        } while (matchPatternMemberRhs != null);
```

Comment:
list match pattern can also be an empty list. So a while loop would be a better option?

Reply:
Changed